Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -23,7 +23,7 @@ export default class VirtualMachineScaleSet {
         var client = new armCompute.ComputeManagementClient(this.taskParameters.credentials, this.taskParameters.subscriptionId);
         var result = await this._getResourceGroupForVmss(client);
         var resourceGroupName: string = result.resourceGroupName;
-        var osType: string = result.osType;
+        var osType: string = this.taskParameters.vmssOsType || result.osType;
```

Comment:
why do you have this || clause. Wont it lead to unpredictability. Or rather when would the vmssOsType would be null?

Reply:
this is for back-compat. currently task works for custom image vmss. for them we will still need to get os type from vmss info.

Reply:
Should we then reverse the order? result.osType || this.taskParameters.vmssOsType

Reply:
yes, can do that.

Reply:
In that case, is it possible to query a data source to find out whether the vmss is custom image and show the input conditionally.
If we are not able to do it, I would suggest, in the help we should clarify that this field is used only when it is gallery image.

Reply:
I would like to keep it simple and ask user. I will add help that this value will be used only in case we case we are not able to detect OS type.
Two reasons:
1.  there might be cases other than gallery image, where we might not detect OS type
2. Querying VMSS is two hop process. first you need to get RG for VMSS and then query for VMSS.

Reply:
Talked to Atul and Muthu. This property will be made mandatory(and always visible). user input will take precedence on auto-detected value. Design time user will see this task as red and will need to mandatory fill it. Runtime back-compat will be maintained.